docs/contributing/03-stories/story-32.md

```diff
@@ -0,0 +1,107 @@
+# User Story 31 - Wing Platforms
+
+Marcus works on a Platform Engineering for a large company that has multiple internal teams all building applications that get deployed into the cloud. 
+
+Recently, his company decided to start investigating adopting Winglang as the main language for their internal teams. However, his company's Cloud Center of Excellence (CCoE) 
+has a few concerns about adopting Winglang. Primarily, they are concerned about how they will still be able to enforce company best practices and standards across all the teams,
+if Wing infers the infrastructure from the code.
+
+Marcus, has taken up the torch to investigate how they can conform the Winglang code written by the internal teams to the company standards. He has stumbled upon the concept of
+Wing Platforms and is now a few energy drinks deep into a late night investigation.
+
+The Wing documentation on Custom Platforms depicts an exposed `interface` called `IPlatform` which defines the methods that a Platform may implement. 
+
+```typescript
+export interface IPlatform {
+  newInstance?(type: string, scope: Construct, id: string, props: any): any;
+  newApp?(): App;
+
+  preSynth?(app: Construct): void;
+  postSynth?(config: any): any;
+  validate?(config: any): any;
+}
+```
+
+As he understands it, Wing comes with a set of built-in platforms `tf-aws`, `tf-azure`, `tf-gcp`, and `sim`. These platforms define how the resources are translated into
+the underlying infrastructure and inflight clients. He knows his company is an AWS shop and many in the company have extensive experience with `Terraform` so the `tf-aws` platform sounds like a great
+place to start.
+
+He has determined that he can create a new platform lovingly dubbed `MarcusPlatform` which should be able to just extend the `tf-aws` platform and override a few methods to
+enforce the company standards.
+
+He begins by creating his platform in a new npm project:
+
+```typescript
+// marcus-platform.ts
+import { Construct } from "constructs";
+import { IPlatform, TfAWS } from "wingsdk/platform";
+
+export class MarcusPlatform extends TfAWS.Platform {
+  newInstance(type: string, scope: Construct, id: string, props: any): any {
+    if (type === "bucket") {
+      // return custom bucket
+      return new Bucket(scope, id, props);
+    }
+
+    return super.newInstance(type, scope, id, props);
+  }
+}
+
+// Custom Bucket class
+export class Bucket extends TfAWS.Bucket {
+  constructor(scope: Construct, id: string, props: any) {
+    super(scope, id, props);
+
+    // enable bucket versioning
+    const bucket = new S3Bucket(scope, name, {
```

Comment:
Yea as is it will not work, unless we store the bucket resource as an instance variable, however many resources create more than 1 thing. The intention is to inherit the bind and inflight methods, but I think I know a cleaner way to model the inheritance

Reply:
maybe `tfaws.Bucket` exposes its internals so they can be configured from the outside:

```js
this.s3Bucket.encryption = ...
```

I think CDKTF resources are mutable.